### PR TITLE
Changes suggested by readability-simplify-boolean-expr fixed

### LINKED ISSUE #15360: clang-tidy: readability-simplify-boolean-expr
Correct changes suggested by clang-tidy: readability-simplify-boolean-expr

### PR DESCRIPTION
**Pull request by LamarMoore**
Resolves #15360 
### Changes

Changes were made to areas of the framework where clang errors were highlighted.
### Testing

Code review. Highlight changes which actually reduce readability of the code.

### CODE REVIEW

File: Framework/DataObjects/src/PeakColumn.cpp [hunk 1]
```diff
@@ -206,8 +206,8 @@ void PeakColumn::read(size_t index, const std::string &text) {
 //-------------------------------------------------------------------------------------
 /** @return true if the column is read-only */
 bool PeakColumn::getReadOnly() const {
-  return !((m_name == "h") || (m_name == "k") || (m_name == "l") ||
-           (m_name == "RunNumber"));
+  return ((m_name == "h") || (m_name == "k") || (m_name == "l") ||
```
Comment (matthew-d-jones) [inline Framework/DataObjects/src/PeakColumn.cpp]: This looks like it returns the opposite bool to before the change, is that intended?

**Comment by LamarMoore** [comment]
@matthew-d-jones  you're correct. Not sure how that happened. I will change it and add a unit test.

File: MantidQt/MantidWidgets/src/InstrumentView/InstrumentActor.cpp [hunk 2]
```diff
@@ -967,7 +967,7 @@ namespace MantidQt
 		*/
 		bool InstrumentActor::hasMaskWorkspace() const
 		{
-			return m_maskWorkspace ? true : false;
+			return static_cast<bool>(m_maskWorkspace);
```
Comment (matthew-d-jones) [inline MantidQt/MantidWidgets/src/InstrumentView/InstrumentActor.cpp]: I think `return m_maskWorkspace != nullptr;` would be preferable here

File: MantidPlot/src/Mantid/MantidDock.cpp [hunk 3]
```diff
@@ -1311,7 +1311,7 @@ void MantidDockWidget::popupMenu(const QPoint &pos) {
 
     // Tell the button what to listen for and what to do once clicked (if there
     // is anything to connect it will be set to false)
-    if (firstPass == false)
+    if (firstPass)
```
Comment (matthew-d-jones) [inline MantidPlot/src/Mantid/MantidDock.cpp]: Should this be `if (!firstPass)`?

File: MantidPlot/src/PythonScript.cpp [hunk 4]
```diff
@@ -506,14 +500,8 @@ void PythonScript::endStdoutRedirect()
...
-  if(codeObject)
-  {
-    return true;
-  }
-  else
-  {
-    return false;
-  }
+
+  return codeObject != NULL;
```
Comment (matthew-d-jones) [inline MantidPlot/src/PythonScript.cpp]: `nullptr` would be preferred over `NULL` here

File: MantidPlot/src/PythonScript.cpp [hunk 5]
```diff
@@ -771,18 +759,7 @@ PyObject* PythonScript::executeCompiledCode(PyObject *compiledCode)
...
-  {
-    return true;
-  }
-  else
-  {
-    return false;
-  }
-}
-
+bool PythonScript::checkResult(PyObject *result) { return result != NULL; }
```
Comment (matthew-d-jones) [inline MantidPlot/src/PythonScript.cpp]: `nullptr`

File: MantidPlot/src/Spectrogram.cpp [hunk 6]
```diff
@@ -583,20 +583,7 @@ void Spectrogram::setLabelsFont(const QFont& font)
...
-            return true;
-        else
-            return false;
-    }
-    return false;*/
-
-  if (d_selected_label)
-    return true;
-  return false;
+  return d_selected_label != NULL;
```
Comment (matthew-d-jones) [inline MantidPlot/src/Spectrogram.cpp]: `nullptr`

File: MantidQt/CustomInterfaces/src/Indirect/IndirectDiffractionReduction.cpp [hunk 7]
```diff
@@ -594,13 +594,13 @@ bool IndirectDiffractionReduction::validateRebin() {
     m_uiForm.valRebinStart->setText("");
     m_uiForm.valRebinWidth->setText("");
     m_uiForm.valRebinEnd->setText("");
-  } else {
+  }
+  else {
 #define CHECK_VALID(text, validator)                                           \
+  rebinValid = !text.isEmpty();												   \
```
Comment (matthew-d-jones) [inline MantidQt/CustomInterfaces/src/Indirect/IndirectDiffractionReduction.cpp]: Now there is an extra call to `test.isEmpty()`, I think it is better left how it was

File: MantidQt/CustomInterfaces/src/Indirect/IndirectMoments.cpp [hunk 8]
```diff
@@ -101,10 +101,9 @@ namespace CustomInterfaces
     if (!msg.isEmpty())
     {
       emit showMessageBox(msg);
-      return false;
     }
 
-    return true;
+    return msg.isEmpty();
```
Comment (matthew-d-jones) [inline MantidQt/CustomInterfaces/src/Indirect/IndirectMoments.cpp]: An extra call to `isEmpty`, better how it was?

File: MantidQt/CustomInterfaces/src/Indirect/JumpFit.cpp [hunk 9]
```diff
@@ -87,10 +87,9 @@ bool JumpFit::validate() {
   QString errors = uiv.generateErrorMessage();
   if (!errors.isEmpty()) {
     emit showMessageBox(errors);
-    return false;
   }
 
-  return true;
+  return errors.isEmpty();
```
Comment (matthew-d-jones) [inline MantidQt/CustomInterfaces/src/Indirect/JumpFit.cpp]: An extra call to `isEmpty`, better how it was?

File: MantidQt/CustomInterfaces/src/Indirect/ResNorm.cpp [hunk 10]
```diff
@@ -99,10 +99,9 @@ bool ResNorm::validate() {
   errors.append(uiv.generateErrorMessage());
   if (!errors.isEmpty()) {
     emit showMessageBox(errors);
-    return false;
   }
 
-  return true;
+  return errors.isEmpty();
```
Comment (matthew-d-jones) [inline MantidQt/CustomInterfaces/src/Indirect/ResNorm.cpp]: An extra call to `isEmpty`, better how it was?

File: MantidQt/CustomInterfaces/src/Indirect/Stretch.cpp [hunk 11]
```diff
@@ -70,10 +70,9 @@ bool Stretch::validate() {
   QString errors = uiv.generateErrorMessage();
   if (!errors.isEmpty()) {
     emit showMessageBox(errors);
-    return false;
   }
 
-  return true;
+  return errors.isEmpty();
```
Comment (matthew-d-jones) [inline MantidQt/CustomInterfaces/src/Indirect/Stretch.cpp]: An extra call to `isEmpty`, better how it was?

**Comment by matthew-d-jones** [comment]
No changes to documentation are necessary for this PR. Code has been carefully reviewed and now looks good.
:shipit:

**Pull request merged**